### PR TITLE
Enhanced RDoc for Float

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #>
- #>=
- #<
- #<=
- #eql?
- #nan?
- #infinite?
- #finite?

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
@jeremyevans, the "What's Here" for Integer will follow this PR; then I'll return to the methods in Float.

File: numeric.c [hunk 1]
```diff
@@ -1671,13 +1671,20 @@ rb_float_cmp(VALUE x, VALUE y)
...
+ *    2.0 > 1.0            # => true
+ *    2.0 > Rational(1, 2) # => true
+ *    2.0 > 2.0            # => false
  *
- * The result of <code>NaN > NaN</code> is undefined,
- * so an implementation-dependent value is returned.
+ *  <tt>Float::NAN > Float::NAN</tt> returns an implementation-dependent value.
+ *
+ *  Raises an exception if +other+ does not respond to method <tt>:></tt>
+ *  (including when +other+ is Complex).
```
Comment (jeremyevans) [inline numeric.c:1687]: I don't think that is correct, at least not in all cases:

```ruby
Rational.undef_method(:>=)
Rational.undef_method(:>)
Rational.undef_method(:<=)
Rational.undef_method(:<)
Rational.undef_method(:<=>)
2.0 >= Rational(1, 2) # => true
2.0 > Rational(1, 2) # => true
2.0 <= Rational(1, 2) # => false
2.0 < Rational(1, 2) # => false
```

How it behaves is going to depend in some cases on `coerce`:

```ruby
class A; def coerce(j); [j, 1] end; end
A.new.respond_to?(:>) # => false
2.0 < A.new # => false
0.0 < A.new # => true
```

I would just drop this paragraph from the documentation from `>`, `>=`, `<`, and `<=`.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good, though I would drop some of the new paragraphs, see inline.

File: numeric.c (hunk 1 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**